Common/src/interface_interpolation/CInterpolator.cpp

```diff
@@ -171,110 +171,117 @@ void CInterpolator::ReconstructBoundary(unsigned long val_zone, int val_marker){
+  map<unsigned long, forward_list<unsigned long>*> neighbors;
 
+  /*--- Define or initialize them. ---*/
   for (iVertex = 0; iVertex < nVertex; iVertex++) {
-
-    Buffer_Send_nLinkedNodes[iVertex] = 0;
-    Aux_Send_Map[iVertex]             = nullptr;
-
     iPoint = geom->vertex[val_marker][iVertex]->GetNode();
-
     if (geom->nodes->GetDomain(iPoint)) {
-      Buffer_Send_GlobalPoint[nLocalVertex] = geom->nodes->GetGlobalIndex(iPoint);
-
+      unsigned long iLocalVertex = iVertex_to_iLocalVertex[iVertex];
+      Buffer_Send_GlobalPoint[iLocalVertex] = (long) geom->nodes->GetGlobalIndex(iPoint);
       for (iDim = 0; iDim < nDim; iDim++)
-        Buffer_Send_Coord[nLocalVertex*nDim+iDim] = geom->nodes->GetCoord(iPoint, iDim);
-
-      nNodes = 0;
-      nEdges = geom->nodes->GetnPoint(iPoint);
-
-      for (jEdge = 0; jEdge < nEdges; jEdge++){
-        EdgeIndex = geom->nodes->GetEdge(iPoint, jEdge);
-
-        if( iPoint == geom->edges->GetNode(EdgeIndex,0) )
-          dPoint = geom->edges->GetNode(EdgeIndex,1);
-        else
-          dPoint = geom->edges->GetNode(EdgeIndex,0);
-
-        if ( geom->nodes->GetVertex(dPoint, val_marker) != -1 )
-          nNodes++;
-      }
-
-      Buffer_Send_StartLinkedNodes[nLocalVertex] = nLocalLinkedNodes;
-      Buffer_Send_nLinkedNodes[nLocalVertex]     = nNodes;
-
-      nLocalLinkedNodes += nNodes;
-
-      Aux_Send_Map[nLocalVertex] = new unsigned long[ nNodes ];
-      nNodes = 0;
-
-      for (jEdge = 0; jEdge < nEdges; jEdge++){
-        EdgeIndex = geom->nodes->GetEdge(iPoint, jEdge);
-
-        if( iPoint == geom->edges->GetNode(EdgeIndex,0) )
-          dPoint = geom->edges->GetNode(EdgeIndex,1);
-        else
-          dPoint = geom->edges->GetNode(EdgeIndex,0);
+        Buffer_Send_Coord[iLocalVertex*nDim+iDim] = geom->nodes->GetCoord(iPoint, iDim);
+      neighbors.insert(pair<unsigned long, forward_list<unsigned long>*>(iPoint, new forward_list<unsigned long>));
+    }
+  }
 
-        if ( geom->nodes->GetVertex(dPoint, val_marker) != -1 ){
-          Aux_Send_Map[nLocalVertex][nNodes] = geom->nodes->GetGlobalIndex(dPoint);
-          nNodes++;
+  /*--- Define the neighbors map. ---*/
+  for(unsigned long iElem=0; iElem < nElems; iElem++){
+    CPrimalGrid* elem = geom->bound[val_marker][iElem];
+    for(unsigned short iNode=0; iNode<elem->GetnNodes(); iNode++){
+      iPoint = elem->GetNode(iNode);
+      if (geom->nodes->GetDomain(iPoint)) {
+        forward_list<unsigned long>* neighb = neighbors.at(iPoint);
+        for(unsigned short iNeighbor=0; iNeighbor<elem->GetnNeighbor_Nodes(iNode); iNeighbor++){
+          unsigned long jPoint = elem->GetNode( elem->GetNeighbor_Nodes(iNode,iNeighbor) );
+          unsigned long jPoint_global = geom->nodes->GetGlobalIndex(jPoint);
+          if( std::find(std::begin(*neighb), std::end(*neighb), jPoint_global) == std::end(*neighb) ){
+            neighb->emplace_front( jPoint_global );
```

Comment:
I think what you want is an unordered_map of unordered_sets...
Linear searches over a list are reallyyyyy expensive, ask me how I know.
Also if you always `emplace_front` you would want a stack, which is a vector upside down.

Reply:
I changed the type of `neighbors` from `map<unsigned long, forward_list<unsigned long>*>` to `unordered_map<unsigned long, unordered_set<unsigned long> >` and `emplace_front` by `insert`, in commit 480d9a3.

Reply:
@maxaehle please dont forget to ask pedro `how he knows`. I would be interested in that